### DISCUSSION TITLE
SDK 1.9 Notice type 18 - Missing schematron rules (BT-14-Lot + BT-5421-Lot, BT-5422-Lot, BT-5423-Lot)

### DISCUSSION THREAD

**Discussion by luispozoWS** [OP-TED/eForms-SDK#818] [Q&A] [answered]
Hello,
Using our own application we have two problems in notice type 18. 
XSD errors appear in two cases despite no schematron rules indicating that these fields are mandatory:

- No schematron rule indicates that BT-14-Lot is mandatory. 
- No schematron rule indicates that at least one of three fields (BT-5421-Lot, BT-5422-Lot or BT-5423-Lot) should be filled.

![AwardCriterionParameter](https://github.com/OP-TED/eForms-SDK/assets/158502029/3f9f4aea-3d98-4cdc-b2a4-f4ee69075fb6)

![Attachment](https://github.com/OP-TED/eForms-SDK/assets/158502029/f69e6c57-8f0e-4d6c-a8f5-ba069d5c9967)

How to deal with this ?

Best regards

**Comment by bertrand-lorentz** [comment]
Could you please indicate which version of the SDK you are using ?
This would help us better understand your issue, as we have added rules in the various SDK versions.

**Reply by luispozoWS** [reply]
Yes, we are currently using SDK 1.9.0

**Comment by bertrand-lorentz** [comment, marked as answer]
1/ For BT-14-Lot
The element required by the schema under `cac:CallForTendersDocumentReference` is `cbc:ID`, which corresponds to OPT-140-Lot.
In the rules, OPT-140-Lot is mandatory if BT-14-Lot is present. And BT-14-Lot is mandatory for subtypes 16, 17 and 19, but not 18. We'll look into this.

2/ For BT-5421-Lot, BT-5422-Lot or BT-5423-Lot
The element required by the schema under `efac:AwardCriterionParameter` is `efbc:ParameterNumeric`, which corresponds to BT-541-Lot-WeightNumber, BT-541-Lot-FixedNumber or BT-541-Lot-ThresholdNumber.
The fields BT-5421-Lot, BT-5422-Lot, BT-5423-Lot are mandatory when the corresponding BT-541-Lot-* field is present. This reflects the fact that when you indicate a number in an award criterion, you must indicate what type of number it is (weight, fixed or threshold).
In the rules, the information on award criterion is not mandatory for notice subtype 18 (so it is not required to indicate a specific number). So it is OK for those fields (BT-541-Lot-* and BT-542x-Lot) to not be present. In this case, the element `efac:AwardCriterionParameter` must be omitted from the XML, as it would be empty.

I hope this helps.
